### PR TITLE
PERF: Refactor next_non_holiday_weekday in XTKS

### LINKED ISSUE #158: PERF: XTKS to load faster
`next_non_holiday_weekday` in `trading_calendar.xtks_holidays` is computationally expensive as observed in #155

### PR DESCRIPTION
**Pull request by leonarduschen**
Closes #158 

Changes:
- 
- Create `sunday_to_tuesday` and `sunday_to_wednesday`
- Split `ConstitutionMemorialDay` into `ConstitutionMemorialDayThrough2006` and `ConstitutionMemorialDay2007Onwards` 
- Delete `is_holiday_or_weekend` and `next_non_holiday_weekday` due to slow loading times, replaced the above rule by applying `sunday_to_tuesday` and `sunday_to_wednesday` to `ConstitutionMemorialDayThrough2006` and `ConstitutionMemorialDay2007Onwards` respectively

### CODE REVIEW

File: trading_calendars/xtks_holidays.py [hunk 1]
```diff
@@ -11,28 +11,24 @@
...
 
 from .common_holidays import new_years_day
-from .trading_calendar import SATURDAY, SUNDAY
+from .trading_calendar import SUNDAY
 
 
-def weekend_to_tuesday(dt):
+def sunday_to_tuesday(dt):
     """
     If holiday falls on Sunday or Saturday, use Tuesday instead.
```
Comment (gerrymanoim) [inline trading_calendars/xtks_holidays.py:19]: Could you update this docstring to reflect your update?

File: trading_calendars/xtks_holidays.py [hunk 2]
```diff
@@ -11,28 +11,24 @@
...
         return dt + timedelta(2)
-    elif dt.weekday() == SATURDAY:
-        return dt + timedelta(3)
     return dt
 
 
-def weekend_to_wednesday(dt):
+def sunday_to_wednesday(dt):
     """
     If holiday falls on Sunday or Saturday, use Wednesday instead.
```
Comment (gerrymanoim) [inline trading_calendars/xtks_holidays.py:28]: Could you update this docstring to reflect your update?

**Comment by gerrymanoim** [comment]
Two small documentation requests, but otherwise looks good to me.

**Review by gerrymanoim — APPROVED** [review]

File: trading_calendars/xtks_holidays.py (hunk 2 above)
Comment (leonarduschen) [reply trading_calendars/xtks_holidays.py]: all done!

**Pull request merged**